File: 0x06-starwars_api/0-starwars_characters.js
#!/usr/bin/node
// StarWars API

const request = require('request');

// Get the movie ID from the command line arguments
const movieId = process.argv[2];

if (!movieId) {
  console.error('Please provide a movie ID.');
  process.exit(1);
}

// URL for the Star Wars API film endpoint with the given movie ID
const apiUrl = `https://swapi-api.alx-tools.com/api/films/${movieId}/`;

// Function to fetch character data
const fetchCharacterName = (url) => {
  return new Promise((resolve, reject) => {
    request(url, (error, response, body) => {
      if (error) {
        reject(error);
      } else {
        const characterData = JSON.parse(body);
        resolve(characterData.name);
      }
    });
  });
};

// Fetch the movie details
request(apiUrl, async (error, response, body) => {
  if (error) {
    console.error('Error fetching movie data:', error);
    return;
  }

  if (response.statusCode === 200) {
    const movieData = JSON.parse(body);
    const characterUrls = movieData.characters;

    // Fetch and print each character's name
    for (const url of characterUrls) {
      try {
        const characterName = await fetchCharacterName(url);
        console.log(characterName);
      } catch (error) {
        console.error('Error fetching character data:', error);
      }
    }
  } else {
    console.error(`Error: Received status code ${response.statusCode}`);
  }
});

File: 0x06-starwars_api/0-starwars_characters.py
#!/usr/bin/env python3

import sys
import requests

def get_movie_characters(movie_id):
    base_url = 'https://swapi-api.alx-tools.com/api'
    film_url = f'{base_url}/films/{movie_id}/'
    response = requests.get(film_url)
    
    if response.status_code != 200:
        print(f"Failed to retrieve movie with ID {movie_id}")
        return
    
    film_data = response.json()
    character_urls = film_data.get('characters', [])
    
    for url in character_urls:
        character_response = requests.get(url)
        if character_response.status_code == 200:
            character_data = character_response.json()
            print(character_data['name'])
        else:
            print(f"Failed to retrieve character data from {url}")

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: ./0-starwars_characters.py <Movie ID>")
        sys.exit(1)
    
    movie_id = sys.argv[1]
    get_movie_characters(movie_id)
